apps/wallet-dashboard/components/Dialogs/SendToken/views/EnterValuesFormView.tsx

```diff
@@ -207,29 +189,43 @@ export function EnterValuesFormView({
         onNext();
     }
 
+    const coinType = coin.coinType;
+    const formattedAmount = parseAmount(formik.values.amount, coinDecimals);
+    const isPayAllIota = formattedAmount === coinBalance && coinType === IOTA_TYPE_ARG;
+
+    const hasEnoughBalance =
+        isPayAllIota ||
+        iotaBalance >
+            BigInt(formik.values.gasBudgetEst ?? '0') +
+                (coinType === IOTA_TYPE_ARG ? formattedAmount : 0n);
+
+    if (coinsBalanceIsPending || coinsIsPending || iotaCoinsIsPending) {
+        return (
+            <div className="flex h-full w-full items-center justify-center">
+                <LoadingIndicator />
+            </div>
+        );
+    }
+
     return (
-        <div className="flex h-full w-full flex-col gap-md">
-            <CoinSelector
-                activeCoinType={coin.coinType}
-                coins={coinsBalance ?? []}
-                onClick={(coinType) => {
-                    setFormData(INITIAL_VALUES);
-                    const coin = coinsBalance?.find((coin) => coin.coinType === coinType);
-                    setSelectedCoin(coin!);
-                }}
-            />
-
-            <Formik
-                initialValues={initialFormValues}
-                validationSchema={validationSchemaStepOne}
-                enableReinitialize
-                validateOnChange={false}
-                validateOnBlur={false}
-                onSubmit={handleFormSubmit}
-            >
-                {(props: FormikProps<FormDataValues>) => (
+        <FormikProvider value={formik}>
+            <DialogLayout>
+                <Header title={'Send'} onClose={onClose} />
+                <DialogLayoutBody>
+                    <CoinSelector
+                        activeCoinType={coin.coinType}
+                        coins={coinsBalance ?? []}
+                        onClick={(coinType) => {
+                            setFormData(INITIAL_VALUES);
+                            const coin = coinsBalance?.find((coin) => coin.coinType === coinType);
+                            setSelectedCoin(coin!);
```

Comment:
avoid using `coin` for 3 different variables

```suggestion
                           const selectedCoin = coinsBalance?.find((coinBalance) => coinBalance.coinType === coinType);
                            if (selectedCoin) {
                                setSelectedCoin(selectedCoin);
                            }
```